components/Buttons/src/TextButton/MDCButton+TextButton.m

```diff
@@ -0,0 +1,36 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCButton+TextButton.h"
+#import "MDCButtonColorThemer.h"
+#import "MDCButtonTypographyThemer.h"
+
+@implementation MDCButton (TextButton)
+
+- (instancetype)initTextButton {
```

Comment:
How will this API work if somebody wants to use a different scheme?

Reply:
i think it is acceptable to offer a convenience initializer for a common use case.  (Perhaps a very common use case)  Although the client will still have to apply their color scheme.
This also acts as an example to client on how they could create their own factory methods.

Reply:
If our intent is to provide examples of factory methods then we should do so in examples or documentation.

This particular implementation bakes in the themes to our component, meaning if we want to migrate clients to a new design system in the future then these defaults will need to be changed (and we'll have to make new APIs).